programs/uxd/src/utils/math/compute_amount_after_change.rs

```diff
@@ -0,0 +1,28 @@
+use crate::error::UxdError;
+use anchor_lang::prelude::*;
+
+pub fn compute_amount_after_change(amount_before_change: u128, change: i128) -> Result<u128> {
+    // In case of a simple positive change (increase), add the two amount
+    if change >= 0 {
+        let increase: u128 = u128::try_from(change).ok().ok_or(UxdError::MathError)?;
+        return Ok(amount_before_change
+            .checked_add(increase)
+            .ok_or(UxdError::MathError)?);
+    }
+    // In case of a negative change, substract using the absolute value (decrease)
+    let decrease: u128 = if change == i128::MIN {
+        // special case: i128::MIN does not have an i128 absolute value
```

Comment:
Has the benefit to cover some edgyedgecase™️

Reply:
yeah that's why i made a utility, to make it very very safe to use accross the codebase :)

Reply:
not possible for a token amount change to hit i128::MIN right?
but still nice to have this as this's a general math util fn

Reply:
ha yhea right..